tests/robot-tests/test_runners.py

```diff
@@ -0,0 +1,110 @@
+import os
+import reports
+import argparse
+import pabot.pabot as pabot
+import robot
+from tests.libs.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def create_robot_arguments(arguments: argparse.Namespace, test_run_folder: str) -> []:
+    robot_args = [
+        "--name",
+        "UI Tests",
+        "--outputdir",
+        f"{test_run_folder}/",
+        "--exclude",
+        "Failing",
+        "--exclude",
+        "UnderConstruction",
+        "--exclude",
+        "VisualTesting",
+        "--xunit",
+        "xunit",
+    ]
+
+    robot_args += ["-v", f"timeout:{os.getenv('TIMEOUT')}", "-v", f"implicit_wait:{os.getenv('IMPLICIT_WAIT')}"]
+    
+    if arguments.fail_fast:
+        robot_args += ["--exitonfailure"]
+    if arguments.tags:
+        robot_args += ["--include", arguments.tags]
+    if arguments.print_keywords:
+        robot_args += ["--listener", "listeners/KeywordListener.py"]
+    if arguments.ci:
+        # NOTE(mark): Ensure secrets aren't visible in CI logs/reports
+        robot_args += ["--removekeywords", "name:operatingsystem.environment variable should be set"]
+        robot_args += ["--removekeywords", "name:common.user goes to url"]  # To hide basic auth credentials
+    
+    process_includes_and_excludes(robot_args, arguments)
```

Comment:
**nitpick** super small nitpick but just wondering if you think something such as `process_environment_args` or something along those lines could be a bit more descriptive for consumers?

Reply:
Hey, so if you're referring to the `process_includes_and_excludes` method name, this is purely for adding the arguments for which tests to include and which to exclude, using the `--include` and `--exclude` options only, hence the name.  I could've left it inline as I only extracted it into its own method for testing some behaviour, but I decided to leave it separate as it does have quite complex logic that I felt was OK to be off on its own.

I will rename it to `_create_include_and_exclude_args` though to make it more private, as it shouldn't be getting used outside of this file really, and got it to return an array of args rather than amending the passed-in args directly.  Thus:

```
    robot_args += _create_include_and_exclude_args(arguments)
```

Reply:
And also put the:

```
if arguments.tags:
        robot_args += ["--include", arguments.tags]
```

part in there too.

Reply:
And:

```
        "--exclude",
        "Failing",
        "--exclude",
        "UnderConstruction",
        "--exclude",
        "VisualTesting",
```